packages/typedoc-plugin-appium/lib/model/reflection/utils.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Adapted from `@knodes/typedoc-pluginutils`
+ * @see https://github.com/knodescommunity/typedoc-plugins
+ * Copyright (c) 2022 KnodesCommunity
+ * Licensed MIT
+ * @see https://github.com/KnodesCommunity/typedoc-plugins/blob/05717565fae14357b1c4be8122f3156e1d46d332/LICENSE
+ */
+
+import {ReflectionKind} from 'typedoc';
+
+const getHigherBitMask = () =>
+  Math.max(
+    ...Object.values({...ReflectionKind, All: -1})
+      .filter((value) => typeof value === 'number')
+      .map((v) => v.toString(2))
+      .filter((v) => v.match(/^0*10*$/))
+      .map((v) => parseInt(v, 2))
+  );
+export const addReflectionKind = (ns: string, name: string, value?: number | null) => {
+  const fullName = `${ns}:${name}`;
+
+  const kindAny = ReflectionKind as any;
+  const existingValue = kindAny[fullName];
+  if (existingValue !== null && existingValue !== undefined) {
+    return existingValue;
+  }
+  const defaultedValue = value ?? getHigherBitMask() * 2;
+  kindAny[fullName] = defaultedValue;
```

Comment:
this is part of the code that I lifted from elsewhere (as noted at the top of the document)

Reply:
i think what hurts my brain is why you are assigning 'defaultedValue' to 'fullName' and 'fullName' to 'defaultedValue'. is this not erroneous?

Reply:
it's a copy-paste job!! not my fault!

Reply:
(it's assigning the `fullName` prop of `kindAny` to a `number` as determined by `getHigherBitMask()` or whatever you passed for `value`. I don't use `value`)

Reply:
i'm still confused. i would expect to see this instead:

```js
kindAny[fullName] = fullName
kindAny[defaultedValue] = defaultedValue
```

anything else seems like a failure of variable naming?

Reply:
I mean, `fullName` is the key and `defaultedValue` is the value.  If you pass a `value`, it uses it, otherwise it computes the next bit mask in sequence.

Reply:
Maybe a failure of variable naming.  I'll just change it to make it more clear

Reply:
i'll take your word for it. for some reason my brain can only read it as a mistake :-)

Reply:
i now get what's going on. sorry.

Reply:
Let me try again.

Enums in TS are both constant types and variables.  There is no way to _add_ stuff to the type of an `enum`; types are immutable.  However, you _can_ treat an `enum` like an object and do whatever you want with it, which is what the author does: `const kindAny = ReflectionKind as any`.

The code as written is dependent on a TS implementation detail (which is unlikely to change); the compiled JS results in an object with its values for its keys and vice-versa.

In other words, if it looks incorrect, it's because it's a hack around the limitations of the TypeDoc API.  It's unfortunate, but necessary.  I had a conversation with a TypeDoc maintainer about this, and think I'll bring it up again...